### PR TITLE
Update script to use .NET 6 build resources

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

- Update the script to use the .NET 6 resources fir building powershell.
- Also consider updated packages on NuGet.org when checking for updates.
- Remove references to .NET 5.
- Add internalfeed url to metadata file

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Review by anmenaga — APPROVED** [review]

File: DotnetRuntimeMetadata.json [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-    "packageVersionPattern": "5.0.0",
-    "sdkImageVersion": "5.0.100",
-    "nextChannel": "net5/rc2"
+    "channel": "release/6.0.1xx-preview1",
+    "packageVersionPattern": "6.0.0-preview.1",
+    "sdkImageVersion": "6.0.100",
+    "nextChannel": "net6/preview1"
+  },
+  "internalfeed" : {
+    "url": ""
```
Comment (rjmholt) [inline DotnetRuntimeMetadata.json:9]: Is this supposed to be empty? If it denotes the absence of a value, `null` might be clearer

File: tools/UpdateDotnetRuntime.ps1 [hunk 2]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
-    } else {
-        'dotnet-internal'
-    }
+    $feedname = if ($UseNuGetOrg) { 'nuget.org' } elseif ($UseInternalFeed) { 'dotnet-internal' } else { 'dotnet' }
 
-    $addDotnet5Source = (-not (Get-PackageSource -Name $feedname -ErrorAction SilentlyContinue))
+    $addDotnetSource = (-not (Get-PackageSource -Name $feedname -ErrorAction SilentlyContinue))
 
-    if (!$UseNuGetOrg -and ($addDotnet5Source -or $addDotnet5InternalSource)) {
+    if (!$UseNuGetOrg -and ($addDotnetSource)) {
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:224]:
```suggestion
    if (!$UseNuGetOrg -and $addDotnetSource) {
```

File: tools/UpdateDotnetRuntime.ps1 [hunk 3]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
         $nugetFileSources = ([xml](Get-Content .\nuget.config -Raw)).Configuration.packagesources.add
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:225]:
```suggestion
        $nugetFileSources = ([xml](Get-Content "$PSScriptRoot/../nuget.config" -Raw)).Configuration.packagesources.add
```

File: tools/UpdateDotnetRuntime.ps1 [hunk 4]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
 
-        if ($addDotnet5Source -and $feedname -ne 'dotnet-internal') {
-            $dotnet5Feed = $nugetFileSources | Where-Object { $_.Key -eq $feedname } | Select-Object -ExpandProperty Value
-            Register-PackageSource -Name $feedname -Location $dotnet5Feed -ProviderName NuGet
+        if ($addDotnetSource -and $feedname -ne 'dotnet-internal') {
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:227]:
```suggestion
        if ($feedname -ne 'dotnet-internal') {
```

File: tools/UpdateDotnetRuntime.ps1 [hunk 5]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
+            $dotnetFeed = $nugetFileSources | Where-Object { $_.Key -eq $feedname } | Select-Object -ExpandProperty Value
+            Register-PackageSource -Name $feedname -Location $dotnetFeed -ProviderName NuGet
             Write-Verbose -Message "Register new package source $feedname" -verbose
         }
 
-        if ($addDotnet5Source -and $InteractiveAuth -and $feedname -eq 'dotnet-internal') {
+        if ($addDotnetSource -and $feedname -eq 'dotnet-internal') {
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:233]:
```suggestion
        if ($feedname -eq 'dotnet-internal') {
```

File: tools/UpdateDotnetRuntime.ps1 [hunk 6]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
             # This NuGet feed is for internal to Microsoft use only.
-            $dotnet5InternalFeed = 'https://pkgs.dev.azure.com/dnceng/internal/_packaging/dotnet5-internal/nuget/v3/index.json'
-            $updatedNugetFile = (Get-Content .\nuget.config -Raw) -replace "</packageSources>", "  <add key=`"dotnet5-internal`" value=`"$dotnet5InternalFeed`" />`r`n  </packageSources>"
+            $dotnetInternalFeed = $dotnetMetadataJson.internalfeed.url
+            $updatedNugetFile = (Get-Content .\nuget.config -Raw) -replace "</packageSources>", "  <add key=`"dotnet6-internal`" value=`"$dotnetInternalFeed`" />`r`n  </packageSources>"
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:236]:
```suggestion
            $updatedNugetFile = (Get-Content "$PSScriptRoot/../nuget.config" -Raw) -replace "</packageSources>", "  <add key=`"dotnet-internal`" value=`"$dotnetInternalFeed`" />`r`n  </packageSources>"
```

File: tools/UpdateDotnetRuntime.ps1 [hunk 7]
```diff
@@ -210,32 +217,26 @@ if ($dotnetUpdate.ShouldUpdate) {
...
             $updatedNugetFile | Out-File .\nuget.config -Force
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:237]: Use `$PSScriptRoot/../nuget.config` or save that path to a variable and use that variable.

Comment (adityapatwardhan) [reply tools/UpdateDotnetRuntime.ps1]: fixed

File: DotnetRuntimeMetadata.json (hunk 1 above)
Comment (adityapatwardhan) [reply DotnetRuntimeMetadata.json]: fixed

**Comment by adityapatwardhan** [comment]
@daxian-dbw @rjmholt Please re-review

**Review by rjmholt — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**